pkg/cm_accounts/cm_accounts.go

```diff
@@ -355,3 +369,52 @@ func (s *service) cmAccount(cmAccountAddr common.Address) (*cmaccount.Cmaccount,
 
 	return cmaccount, nil
 }
+
+func (s *service) getCurrentImplementationOnManager(cmAccountAddress common.Address) (common.Address, error) {
+	cmAccount, err := cmaccount.NewCmaccount(cmAccountAddress, s.ethClient)
+	if err != nil {
+		return common.Address{}, fmt.Errorf("failed to fetch CM account: %w", err)
+	}
+	managerAddress, err := cmAccount.GetManagerAddress(&bind.CallOpts{})
+	if err != nil {
+		return common.Address{}, fmt.Errorf("failed to fetch CM account Manager Address: %w", err)
+	}
+	manager, err := cmaccountmanager.NewCmaccountmanager(managerAddress, s.ethClient)
+	if err != nil {
+		return common.Address{}, fmt.Errorf("failed to get Manager: %w", err)
+	}
+	currentImplOnManager, err := manager.GetAccountImplementation(&bind.CallOpts{})
+	if err != nil {
+		return common.Address{}, fmt.Errorf("failed to get Account Implementation: %w", err)
+	}
+	return currentImplOnManager, nil
+}
+
+func (s *service) getCurrentImplementationOnProxy(ctx context.Context, cmAccountAddress common.Address) (common.Address, error) {
+	implAddress, err := s.ethClient.StorageAt(ctx, cmAccountAddress, implementationSlot, nil)
+	if err != nil {
+		return common.Address{}, fmt.Errorf("failed to get implementation address from proxy: %w", err)
+	}
+	if len(implAddress) < 32 {
+		return common.Address{}, fmt.Errorf("implementation address storage read returned unexpected size: %d", len(implAddress))
+	}
+	currentImplOnProxy := common.BytesToAddress(implAddress[12:])
+	return currentImplOnProxy, nil
+}
+
+func (s *service) IsCmAccountImplementationUpToDate(ctx context.Context, cmAccountAddress common.Address) (bool, error) {
```

Comment:
```suggestion
func (s *service) IsCMAccountImplementationUpToDate(ctx context.Context, cmAccountAddress common.Address) (bool, error) {
```
https://google.github.io/styleguide/go/decisions#initialisms